src/lint/collect-header-diagnostics.ts

```diff
@@ -0,0 +1,113 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { getLocation, indexWithinElementToTrueLocation } from './utils';
+
+const ruleId = 'header-format';
+
+export function collectHeaderDiagnostics(
+  dom: any,
+  headers: { element: Element; contents: string }[]
+) {
+  let lintingErrors: LintingError[] = [];
+
+  for (let { element, contents } of headers) {
+    if (!/\(.*\)$/.test(contents) || / Operator \( `[^`]+` \)$/.test(contents)) {
+      continue;
+    }
+
+    let name = contents.substring(0, contents.indexOf('('));
+    let params = contents.substring(contents.indexOf('(') + 1, contents.length - 1);
+
+    if (!/[\S] $/.test(name)) {
+      let { line, column } = indexWithinElementToTrueLocation(
+        getLocation(dom, element),
+        contents,
+        name.length
+      );
+      lintingErrors.push({
+        ruleId,
+        nodeType: element.tagName,
+        line,
+        column,
+        message: 'expected header to have a single space before the argument list',
+      });
+    }
+
+    let nameMatches = [
+      // Runtime Semantics: Foo
+      /^(Runtime|Static) Semantics: [A-Z][A-Za-z0-9/]*\s*$/,
+
+      // Number::foo
+      /^[A-Z][A-Za-z0-9]*::[a-z][A-Za-z0-9]*\s*$/,
+
+      // [[GetOwnProperty]]
+      /^\[\[[A-Z][A-Za-z0-9]*\]\]\s*$/,
+
+      // _NativeError_
+      /^_[A-Z][A-Za-z0-9]*_\s*$/,
+
+      // CreateForInIterator
+      // Object.fromEntries
+      // Array.prototype [ @@iterator ]
+      /^[A-Za-z][A-Za-z0-9]*(\.[A-Za-z][A-Za-z0-9]*)*( \[ @@[a-z][a-zA-Z]+ \])?\s*$/,
+
+      // %ForInIteratorPrototype%.next
+      // %TypedArray%.prototype [ @@iterator ]
+      /^%[A-Z][A-Za-z0-9]*%(\.[A-Za-z][A-Za-z0-9]*)*( \[ @@[a-z][a-zA-Z]+ \])?\s*$/,
+    ].some(r => r.test(name));
+
+    if (!nameMatches) {
+      let { line, column } = indexWithinElementToTrueLocation(
+        getLocation(dom, element),
+        contents,
+        0
+      );
+      lintingErrors.push({
+        ruleId,
+        nodeType: element.tagName,
+        line,
+        column,
+        message: `expected operation to have a name like 'Example', 'Runtime Semantics: Foo', 'Example.prop', etc, but found ${JSON.stringify(
+          name
+        )}`,
+      });
+    }
+
+    let paramsMatches =
+      params.match(/\[/g)?.length === params.match(/\]/g)?.length &&
+      [
+        // Foo ( )
+        /^ $/,
+
+        // Object ( . . . )
+        /^ \. \. \. $/,
+
+        // String.raw ( _template_, ..._substitutions_ )
+        /^ (_[A-Za-z0-9]+_, )*\.\.\._[A-Za-z0-9]+_ $/,
+
+        // Function ( _p1_, _p2_, &hellip; , _pn_, _body_ )
+        /^ (_[A-Za-z0-9]+_, )*… (, _[A-Za-z0-9]+_)+ $/,
```

Comment:
```suggestion
        /^ (_[A-Za-z0-9]+_, )*…( , _[A-Za-z0-9]+_)* $/,
```

Reply:
I explicitly do not want to allow the form which has `…` without a following argument.

Reply:
What about a preceding argument?

```suggestion
        /^ (_[A-Za-z0-9]+_, )+…( , _[A-Za-z0-9]+_)+ $/,
```

At the very least, you need this one:

```suggestion
        /^ (_[A-Za-z0-9]+_, )*…( , _[A-Za-z0-9]+_)+ $/,
```

Reply:
> What about a preceding argument?

I'm fine with this matching `(..., _a_)`.

>  At the very least, you need this one:

No I don't. That would match `… , _a_ , _b_` instead of `… , _a_, _b_`. I want the second of those.

Reply:
Oops, I meant

```suggestion
        /^ (_[A-Za-z0-9]+_, )*…(, _[A-Za-z0-9]+_)+ $/,
```

Reply:
No, that comma is [there upstream](https://tc39.es/ecma262/#sec-function-p1-p2-pn-body), and I'm just trying to match the current upstream conventions.